zulipterminal/model.py

```diff
@@ -1086,6 +1088,22 @@ def _handle_update_message_flags_event(self, event: Event) -> None:
             set_count(list(message_ids_to_mark & indexed_message_ids),
                       self.controller, -1)
 
+    def _handle_update_emoji_event(self, event: Event) -> None:
+        """
+        Handle update of emoji
+        """
+        realm_emoji_update = event['realm_emoji']
+        # Removes emojis that have been deleted
+        # from the active_emoji_data
+        for emoji_code, emoji in realm_emoji_update.items():
+            if emoji['deactivated']:
+                _ = self.active_emoji_data.pop(emoji['name'], "")
```

Comment:
Note that if the deactivated emoji had the same name as a unicode emoji, then this is the wrong behavior, as it doesn't bring back the unicode emoji with that name.

We could fully regenerate the active emoji in a function, in the first instance, like we do in `__init__`

Reply:
When a realm emoji and unicode emoji have the same name, the unicode emoji is replaced by the realm emoji.
According to what you have mentioned here, the code should bring back this unicode emoji once the user deactivates the realm emoji with the same name.
Correct me if I'm wrong

Reply:
The order goes unicode<realm<zulip, which is why we have the dict unpacking of each individual dict in the way we do, in `__init__`.